addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -658,12 +659,12 @@ function factory(dependencies) {
                     owl.utils.escape(channel.name)
                 );
             }
+            channel.update({ serverPinState: 'unpin' });
```

Comment:
worth a comment that this is necessary because this notification does not contain the new channel_info so we have to assume.